fe/src/main/java/org/apache/doris/rewrite/FEFunctions.java

```diff
@@ -92,415 +89,67 @@ public static DateLiteral daysAdd(LiteralExpr date, LiteralExpr day) throws Anal
-        d = DateUtils.addDays(d, -(int) day.getLongValue());
-        return new DateLiteral(DateFormatUtils.format(d, "yyyy-MM-dd HH:mm:ss"), Type.DATETIME);
+        return dateAdd(date, new IntLiteral(-(int) day.getLongValue()));
     }
 
     @FEFunction(name = "year", argTypes = { "DATETIME" }, returnType = "INT")
     public static IntLiteral year(LiteralExpr arg) throws AnalysisException {
-        long timestamp = getTime(arg);
-        Calendar instance = Calendar.getInstance();
-        instance.setTimeInMillis(timestamp);
-        return new IntLiteral(instance.get(Calendar.YEAR), Type.INT);
+        return new IntLiteral(((DateLiteral) arg).getYear(), Type.INT);
     }
 
     @FEFunction(name = "month", argTypes = { "DATETIME" }, returnType = "INT")
     public static IntLiteral month(LiteralExpr arg) throws AnalysisException {
-        long timestamp = getTime(arg);
-        Calendar instance = Calendar.getInstance();
-        instance.setTimeInMillis(timestamp);
-        return new IntLiteral(instance.get(Calendar.MONTH) + 1, Type.INT);
+        return new IntLiteral(((DateLiteral) arg).getMonth(), Type.INT);
     }
 
     @FEFunction(name = "day", argTypes = { "DATETIME" }, returnType = "INT")
     public static IntLiteral day(LiteralExpr arg) throws AnalysisException {
-        long timestamp = getTime(arg);
-        Calendar instance = Calendar.getInstance();
-        instance.setTimeInMillis(timestamp);
-        return new IntLiteral(instance.get(Calendar.DAY_OF_MONTH), Type.INT);
+        return new IntLiteral(((DateLiteral) arg).getDay(), Type.INT);
     }
 
     @FEFunction(name = "unix_timestamp", argTypes = { "DATETIME" }, returnType = "INT")
     public static IntLiteral unix_timestamp(LiteralExpr arg) throws AnalysisException {
-        long timestamp = getTime(arg);
-        return new IntLiteral(timestamp / 1000, Type.INT);
+        try {
+            return new IntLiteral(((DateLiteral) arg).unixTime(TimeUtils.getTimeZone()) / 1000, Type.INT);
+        } catch (ParseException e) {
+            throw new AnalysisException(e.getLocalizedMessage());
+        }
     }
 
     @FEFunction(name = "from_unixtime", argTypes = { "INT" }, returnType = "VARCHAR")
     public static StringLiteral fromUnixTime(LiteralExpr unixTime) throws AnalysisException {
-        long ts = unixTime.getLongValue();
-        // if unixTime < 0 or larger than 9999-12-31 23:59:59, we should return null, throw a exception and let BE process
-        if (ts < 0 || ts > 253402271999L) {
+        //if unixTime < 0, we should return null, throw a exception and let BE process
+        if (unixTime.getLongValue() < 0) {
             throw new AnalysisException("unixtime should larger than zero");
         }
-        Date date = new Date(unixTime.getLongValue() * 1000);
-        return new StringLiteral(dateFormat(date, "%Y-%m-%d %H:%i:%S"));
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```

Comment:
I think we can create a constructor of DateLiteral with unix timestamp and timezone.
Then this logic can be included in DateLiteral